students/roviram/lesson08/test_circle_class.py

```diff
@@ -0,0 +1,155 @@
+# Dev: Miguel Rovira-Gonzalez
+# Lesson08: Circle Assignment
+
+import pytest
+import math
+from circle_class import Circle, Sphere
+
+
+def test_init():
+    c = Circle(10)
+    assert c.radius == 10
+
+
+def test_no_radius():
+    with pytest.raises(TypeError):
+        c = Circle()
+
+
+def test_diameter():
+    c = Circle(10)
+    assert c.diameter == 20
+
+
+def test_diameter_change_radius():
+    c = Circle(10)
+    assert c.diameter == 20
+    c.radius = 4
+    assert c.radius == 4
+    assert c.diameter == 8
+
+
+def test_diameter_change_diameter():
+    c = Circle(10)
+    assert c.radius == 10
+    assert c.diameter == 20
+
+    c.diameter = 4
+    assert c.radius == 2
+    assert c.diameter == 4
+
+
+def test_area():
+    c = Circle(2)
+    assert math.isclose(c.area, 12.566370, rel_tol=1e-7)
+
+
+def test_str():
+    c = Circle(20)
+    assert str(c) == "Circle with radius: 20"
+
+
+def test_rep():
+    c = Circle(10)
+    assert repr(c) == "Circle10"
+
+
+def test_set_area():
+    c = Circle(5)
```

Comment:
You won't need this error here since you will never throw an AttributeError for Circle. The area method is implemented here.